LeaderboardBackend/Controllers/ModshipsController.cs

```diff
@@ -76,4 +76,43 @@ public async Task<ActionResult> MakeMod([FromBody] CreateModshipRequest body)
 		await _modshipService.CreateModship(modship);
 		return CreatedAtAction(nameof(MakeMod), new { id = modship.Id }, modship);
 	}
+
+	/// <summary>Removes a User as a Mod from a Leaderboard. Admin-only.</summary>
+	/// <param name="body">A RemoveModshipRequest</param>
+	/// <response code="204">Request was successfull.</response>
+	/// <response code="400">If the request is malformed.</response>
+	/// <response code="404">The User, Leaderboard or Modship was not found.</response>
+	[ApiConventionMethod(typeof(Conventions),
+						 nameof(Conventions.Delete))]
+	[Authorize(Policy = UserTypes.Admin)]
+	[HttpDelete]
+	public async Task<ActionResult> RemoveMod([FromBody] RemoveModshipRequest body)
+	{
+		User? user = await _userService.GetUserById(body.UserId);
+		Leaderboard? leaderboard = await _leaderboardService.GetLeaderboard(body.LeaderboardId);
+
+		if (user is null || leaderboard is null)
+		{
+			return NotFound();
+		}
+
+		Modship modship = new()
```

Comment:
@zysim had a good suggestion for this: we should add something to the `ModshipService` that looks up a Modship by UserId and LeaderboardId, then use that here. It makes a lot more sense, because we can check for null here. If this passes a null check, then all the delete has to do is mark it, rather than needing to worry about doing a lookup and throwing an exception.

So if possible:
1. Add a method to the ModshipService like `GetModshipForLeaderboard`
2. Have that method return null if nothing is found
3. Do a null check here and `return NotFound` if it is (similar to other lookup actions)
4. Make `DeleteModship` just do the marking deleted part

I think that would put us in a pretty good spot for this, and would be a good model for future deletes.

Reply:
Good points. Changed as suggested.